applications/ChimeraApplication/custom_strategies/custom_builder_and_solver/residualbased_block_builder_and_solver_with_constraints_for_chimera.h

```diff
@@ -0,0 +1,312 @@
+
+            TSystemVectorType b_modified(rb.size());
+            //boost::numeric::ublas::axpy_prod(boost::numeric::ublas::trans(mL), rb, b_modified, true);
+            TSparseSpace::Mult(L_transpose_matrix, rb, b_modified);
+            TSparseSpace::Copy(b_modified, rb);
+            b_modified.resize(0, false); //free memory
+
+            TSystemMatrixType auxiliar_A_matrix(BaseType::mT.size2(), rA.size2());
+            //boost::numeric::ublas::axpy_prod(boost::numeric::ublas::trans(mL), rA, auxiliar_A_matrix, true);
+            SparseMatrixMultiplicationUtility::MatrixMultiplication(L_transpose_matrix, rA, auxiliar_A_matrix); //auxiliar = T_transpose * rA
+            L_transpose_matrix.resize(0, 0, false);                                                             //free memory
+
+            //boost::numeric::ublas::axpy_prod(auxiliar_A_matrix, BaseType::mT, rA, true);
+            SparseMatrixMultiplicationUtility::MatrixMultiplication(auxiliar_A_matrix, BaseType::mT, rA); //A = auxilar * T   NOTE: here we are overwriting the old A matrix!
+            auxiliar_A_matrix.resize(0, 0, false);                                                        //free memory
+
+            double max_diag = 0.0;
+            for (IndexType i = 0; i < rA.size1(); ++i)
+            {
+                max_diag = std::max(std::abs(rA(i, i)), max_diag);
+            }
+// Apply diagonal values on slaves  BaseType::mDofSet.size()
+#pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(BaseType::mSlaveIds.size()); ++i)
+            {
+                const IndexType slave_equation_id = BaseType::mSlaveIds[i];
+                if (BaseType::mInactiveSlaveDofs.find(slave_equation_id) == BaseType::mInactiveSlaveDofs.end())
+                {
+                    rA(slave_equation_id, slave_equation_id) = max_diag;
+                    rb[slave_equation_id] = 0.0;
+                }
+            }
+            const double stop_constraints = OpenMPUtils::GetCurrentTime();
+            KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Applying constraints time: " << stop_constraints - start_constraints << std::endl;
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    ///@}
+
```

Comment:
I think you can remove the private section.

Reply:
I can .. This comes with the default format of the files. :-|